### PR TITLE
Add min delay metrics

### LINKED ISSUE #5618: SAE metrics bucket 6: minimum block delay
SAE metrics bucket 6: minimum block delay

| Metric | Type | Operational question |
| --- | --- | --- |
| `min_block_delay_seconds` | Gauge | What minimum block delay is in force? |

## Metric design

- **Single gauge, not an `accepted`/`executed` pair.** The ACP-226 delay excess is a control parameter set by the header and steered by validator votes; execution never mutates it. An accepted and an executed value would report the same number offset only by backlog lag, so one gauge suffices. This matches `gas_target` in bucket 4 and contrasts with `base_fee`/`gas_excess`, which execution does move.
- **Lives in `saexec`, updated at `markEnqueued`** from the enqueued block's header — the same site and header source `accepted_gas_limit_total` reads. `DelayExponent.Delay()` returns milliseconds; divide by 1000 for the base-unit `_seconds` value.
- **Not derivable from observed block cadence.** Realized inter-block time is the actual spacing; this gauge is the enforced floor, and it is consensus state, not any single node's config.

Implementation follows the per-package unexported `metrics` struct pattern established in #5362.

### PR DESCRIPTION
**Pull request by rahulmutt-ava**
## Why this should be merged

Resolves #5618. Adds metrics for acp-226 current minimum block delay.

## How this works

Creates a new `hook.Points` method to return the minimum block delay from the supplied header otherwise we leak c-chain specific headers into sae generally and then propagates the block delay metric after execution is done.

## How this was tested
Added a metrics unit test.

## Need to be documented in RELEASES.md?
Yes, the new metric name should be listed.

### CODE REVIEW

**Comment by JonathanOppenheimer** [comment]
As discussed in our standup, I think this can all just be on the `AfterExecutingblock` hook -- a slight delay isn't worth all the machinery!

File: vms/saevm/cchain/metrics_test.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+	"testing"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/stretchr/testify/require"
+
+	"github.com/ava-labs/avalanchego/vms/saevm/cchain/tx/txtest"
+)
+
+// gaugeValue gathers metricName from g and returns its single gauge sample.
+func gaugeValue(tb testing.TB, g prometheus.Gatherer, metricName string) float64 {
```
Comment (JonathanOppenheimer) [inline vms/saevm/cchain/metrics_test.go]: I don't think we need this whole helper -- you could drop in `testutil.GatherAndCompare`

Comment (JonathanOppenheimer) [reply vms/saevm/cchain/metrics_test.go]: Additionally, we decided that we did not need to test these metrics at all -- you'll see that none of the cchain metrics have tests. So unless this one is special, I would just delete this whole file.

Comment (rahulmutt-ava) [reply vms/saevm/cchain/metrics_test.go]: good point on trying to use the `testutil` functions as much as possible - I restructured a bit so I can just use `testutil.ToFloat64`.

I'd argue that if we're listing metrics in `RELEASES.md` they're something consumed by the user of this software and should be treated as a form of validator / operator API and therefore should have tests when possible to make sure the metric is measuring what we think it's measuring. I'd expect every validator has their own Grafana-like setup using the prometheus metrics to keep track of the node health (among other use cases they might have).

I will do a followup PR for testing the existing SAE metrics once all the metrics are in.

Comment (JonathanOppenheimer) [reply vms/saevm/cchain/metrics_test.go]: We talked about this fairly extensively after I included tests on my first SAE metrics PR. I don't think testing metrics adds much value at all regardless, but if we want to revisit that decision, we should!

**Review by JonathanOppenheimer — APPROVED** [review]

File: vms/saevm/cchain/metrics_test.go (hunk 1 above)
Comment (rahulmutt-ava) [reply vms/saevm/cchain/metrics_test.go]: yeah we should have another discussion. To be clear I'm not saying that we should take it as seriously as tests for actual chain / protocol consistency and to prioritize this over other things, but I do feel we need to have a test for every user-facing functionality that we explicitly document.

File: RELEASES.md [hunk 2]
```diff
@@ -14,6 +14,7 @@
   - `avalanche_{vmName}_sae_executed_gas_limit_total` (counter): cumulative gas limit (worst-case gas) of executed blocks.
 - Added `avalanche_{vmName}_sae_in_memory_blocks` (gauge): number of SAE blocks still live in memory (created but not yet garbage collected).
 - Added `avalanche_{vmName}_sae_accepted_gas_limit_total` (counter): cumulative gas limit (worst-case gas) of blocks accepted into the execution queue; the acceptance-side counterpart of `executed_gas_limit_total`.
+- Added `avalanche_{vmName}_cchain_min_block_delay_seconds` (gauge): ACP-226 minimum block delay currently in force, taken from the most recently executed block.
```
[the remark this answers is not in the document]
Comment (rahulmutt-ava) [reply RELEASES.md:25]: this is intended

File: vms/saevm/cchain/metrics.go [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+	minBlockDelay prometheus.Gauge
+}
+
+func newMetrics(reg prometheus.Registerer) (*metrics, error) {
+	m := &metrics{
+		minBlockDelay: prometheus.NewGauge(prometheus.GaugeOpts{
+			Name: "min_block_delay_seconds",
+			Help: "ACP-226 minimum block delay, in seconds.",
+		}),
+	}
```
Comment (alarso16) [inline vms/saevm/cchain/metrics.go:22]: What other metrics are we going to add? Just making sure this is scoped correctly, because right now it only needs to be used by the hooks

**Review by alarso16 — APPROVED** [review]

**Review by powerslider — APPROVED** [review]

File: vms/saevm/cchain/vm.go [hunk 4]
```diff
@@ -121,6 +122,16 @@ func (vm *VM) Initialize(
...
 
+	reg, err := apimetrics.MakeAndRegister(snowCtx.Metrics, "cchain")
+	if err != nil {
+		return fmt.Errorf("making metrics: %w", err)
+	}
+	m, err := newMetrics(reg)
+	if err != nil {
+		return fmt.Errorf("registering cchain metrics: %w", err)
+	}
+	vm.metrics = m
```
Comment (StephenButtolph) [inline vms/saevm/cchain/vm.go]: This aligns a bit more with how the rest of the function initializes stuff.
```suggestion
	vm.metrics, err = newMetrics(reg)
	if err != nil {
		return fmt.Errorf("registering cchain metrics: %w", err)
	}
```

**Review by StephenButtolph — APPROVED** [review]
trivial nit then lgtm
